src/main/java/org/broadinstitute/dsm/route/PatchRoute.java

```diff
@@ -303,13 +305,29 @@ private void writeDSMRecordsToES(@NonNull Patch patch) {
     private void writeESWorkflow(@NonNull Patch patch, @NonNull NameValue nameValue, @NonNull Value action) {
         DDPInstance ddpInstance = DDPInstance.getDDPInstance(patch.getRealm());
         String status = nameValue.getValue() != null ? String.valueOf(nameValue.getValue()) : null;
-        if (StringUtils.isNotBlank(status)) {
-            Map<String,String> data = new Gson().fromJson(status, new TypeToken<Map<String, String>>(){}.getType());
-            if (action.getValue() != null && StringUtils.isNotBlank(action.getValue())) {
-                ElasticSearchUtil.writeWorkflow(ddpInstance, patch.getParentId(), action.getName(), action.getValue());
+        if (StringUtils.isBlank(status)) {
+            return;
+        }
+        Map<String,String> data = new Gson().fromJson(status, new TypeToken<Map<String, String>>(){}.getType());
+        if (action.getValue() != null && StringUtils.isNotBlank(action.getValue())) {
+            if (patch.getFieldId().contains(FamilyMemberConstants.GROUP)) {
+                ElasticSearchUtil.writeWorkflow(WorkflowForES.createInstance(ddpInstance, patch.getParentId(), action.getName(), action.getValue()));
+            } else if (ParticipantDataUtil.hasProbandEmail(patch.getParentId(), data.get(FamilyMemberConstants.COLLABORATOR_PARTICIPANT_ID))) {
```

Comment:
What's the point of using comma separator in `if` statement? As I know first result will be ignored and are you sure that you want it like that? (or comma separator in `if` statement works differently for java?)